File: 2023-06-20-08-28-07/problems/find_the_highest_altitude/solution.py
class Solution:
    def largestAltitude(self, gain: List[int]) -> int: 
        n = len(gain)
        prefixSum = [0] * (n + 1)  # Initialize prefix sum array
        maxAltitude = 0

        for i in range(1, n + 1):
            prefixSum[i] = prefixSum[i - 1] + gain[i - 1]
            maxAltitude = max(maxAltitude, prefixSum[i])

        return maxAltitude

File: 2023-06-20-08-28-07/problems/find_the_highest_altitude/solution.java
class Solution {
    public int largestAltitude(int[] gain) {
        int n = gain.length;
        int[] prefixSum = new int[n + 1];
        int maxAltitude = 0;

        for (int i = 1; i <= n; i++) {
            prefixSum[i] = prefixSum[i - 1] + gain[i - 1];
            if (prefixSum[i] > maxAltitude) {
                maxAltitude = prefixSum[i];
            }
        }

        return maxAltitude;
    }
}